frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/ApplicationMaster.java

```diff
@@ -321,18 +322,11 @@ private void stopForInternalUnKnownError(String customizedDiagnostics) {
     stop(new StopStatus(ExitStatusKey.AM_INTERNAL_UNKNOWN_ERROR.toInt(), true, diagnostics));
   }
 
-  // Principle to setup ContainerRequest for a Task:
-  // 1. Exactly match the Task's Requirement
-  //    -> Keeps Waiting Allocation, i.e. containerRequestTimeoutSec = -1
-  // 2. Too Relax for the Task's Requirement
-  //    -> Reject Allocation and Re-Request, See testContainer
-  // 3. Too Strict for the Task's Requirement
-  //    -> Timeout Request and Re-Request, i.e. containerRequestTimeoutSec != -1
```

Comment:
Why remove it

Reply:
The logic totally different, it is simple now. 

---
In reply to: [173126589](https://github.com/Microsoft/pai/pull/318#discussion_r173126589) [](ancestors = 173126589)

Reply:
But the Principle is not changed 

---
In reply to: [173372505](https://github.com/Microsoft/pai/pull/318#discussion_r173372505) [](ancestors = 173372505,173126589)